framework/include/utils/IndexableProperty.h

```diff
@@ -123,32 +123,40 @@ IndexableProperty<T, is_ad>::operator[](int qp) const
 }
 
 template <typename T, bool is_ad>
-IndexableProperty<T, is_ad>::operator bool() const
+void
+IndexableProperty<T, is_ad>::check() const
 {
-  return _property_real || _property_std_vector || _property_real_vector_value ||
-         _property_rank_two_tensor || _property_rank_three_tensor || _property_rank_four_tensor;
+  if (_property_real)
+    checkComponents(0);
+  else if (_property_std_vector)
+    checkComponents(1);
+  else if (_property_real_vector_value)
+    checkComponents(1);
+  else if (_property_rank_two_tensor)
+    checkComponents(2);
+  else if (_property_rank_three_tensor)
+    checkComponents(3);
+  else if (_property_rank_four_tensor)
+    checkComponents(4);
+  else
+    mooseError("The ",
```

Comment:
We should probably change all of the errors in here to `_host->mooseError()` and make `_host` be a `MooseObject`. That way we can get some more error context